bin/vsm.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# Copyright (C) 2017 Collabora
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Authors: Shane Fagan - shane.fagan@collabora.com
+
+import datetime
+import sys
+import os
+import argparse
+import yaml
+import glob
+import ast
+from time import sleep
+
+LOGIC_REPLACE = {'||': 'or',
+                 '&&': 'and',
+                 'true': '1',
+                 'false': '0'}
+
+KEYWORDS = ['condition', 'emit']
+
+class State():
+    '''
+        Class to handle states
+    '''
+    def __init__(self, conf, rules_dir):
+        with open(conf) as f:
+            data = yaml.load(f.read())
+
+        for item in data:
+            item = item.replace(" ", "").split("=")
+            vars(self)[item[0]] = item[1]
+
+        self.code = dict()
+        for file_name in glob.glob(os.path.join(rules_dir, '*.yaml')):
+            with open(file_name) as f:
+                sig_name = file_name.strip(".yaml").split("/")[-1]
+                self.code[sig_name] = parse_rules(f.read())
+
+
+def __process_rules(data):
```

Comment:
So now you have duplicated this from the rules parser, you can just remove that file, then. Btw, why the __ prefix?